src/trackinfoobject.h

```diff
@@ -413,7 +370,14 @@ class TrackInfoObject : public QObject {
     CoverArt m_coverArt;
 
     friend class TrackDAO;
-    friend class AutoDJProcessorTest;
+    friend class LegacyLibraryImporter;
+    friend class SoundSourceProxy;
 };
 
-#endif
+// The following functions will create a temporary SoundSourceProxy
+// object internally. Please use the corresponding member functions
+// from SoundSourceProxy if possible.
+void parseTrackMetadata(TrackPointer pTrack);
```

Comment:
We typically avoid global scope functions in Mixxx. Could you make these static methods of TrackInfoObject?